pygmt/_show_versions.py

```diff
@@ -38,6 +39,31 @@ def _get_module_version(modname: str) -> str | None:
         return None
 
 
+def _get_gdal_version() -> str | None:
+    """
+    Get GDAL version by calling the GDAL C API via ctypes.
+    """
+    match sys.platform:
+        case name if name == "linux" or name.startswith("freebsd"):
+            libname = "libgdal.so"
+        case "darwin":
+            libname = "libgdal.dylib"
+        case "win32":
+            libname = "gdal.dll"
+        case _:
+            return None
+
+    try:
+        lib = ctypes.CDLL(libname)
+        lib.GDALVersionInfo.restype = ctypes.c_char_p
+        version = lib.GDALVersionInfo(b"RELEASE_NAME")  # e.g., 3.6.3
+        if version is not None:
+            return version.decode("utf-8")
+    except (OSError, AttributeError):
+        return None
+    return None
```

Comment:
This last return statement is probably not needed? Unless there is a chance that something other than an OSError or AttributeError is raised?

```suggestion
```